### PR TITLE
Sort generated numbers

### PR DESCRIPTION
**Pull request by adhrinae**
Usually, when I buy lotto, the lottery numbers are automatically sorted from the smallest to the largest.
So It would be useful if this application provides sorted numbers.

### CODE REVIEW

File: src/test.ts [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
+
+  const [head, ...rest] = list;
+  return recursiveSortCheck(head, rest);
+}
+
+test('isSortedList', async t => {
+  const sortedList = [1, 2, 3, 4, 5, 6]
+  const notSortedList = [7, 10, 2, 58, 1, 40]
+
+  t.is(isSortedList(sortedList), true);
```
Comment (ragingwind) [inline src/test.ts]: IMO, `generateNumber().every(isSorted)` is enough though

File: src/test.ts [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+  t.is(isSortedList(notSortedList), false);
+})
+
+
+test('generateNumber', async t => {
+  const defaultLength = generateNumber();
+  const longerNumbers = generateNumber(10);
+  const lonegerNumbersWithLimit = generateNumber(10, 45);
+
+  t.is(isSortedList(defaultLength), true);
```
Comment (ragingwind) [inline src/test.ts]: dup, we already have it above. checking length only

File: src/test.ts [hunk 3]
```diff
@@ -1,13 +1,43 @@
...
+  const [head, ...rest] = list;
+  return recursiveSortCheck(head, rest);
+}
+
+test('isSortedList', async t => {
+  const sortedList = [1, 2, 3, 4, 5, 6]
+  const notSortedList = [7, 10, 2, 58, 1, 40]
+
+  t.is(isSortedList(sortedList), true);
+  t.is(isSortedList(notSortedList), false);
```
Comment (ragingwind) [inline src/test.ts]: ^ t.false or true

File: src/test.ts (hunk 2 above)
Comment (adhrinae) [reply src/test.ts]: But if I test only length of `generateNumber()`, how can I be sure the generated numbers are sorted in the test case?

Comment (ragingwind) [reply src/test.ts]: ~~Sorting test is added from your PR~~. (I think sorting test is added at first) testing should be isolated and separated. You should add a new test for sorting not updated length test even it was not tagged.

test('sorting' ...)
test('length' ...)
test('normalize' ...)

File: src/test.ts [hunk 4]
```diff
@@ -1,39 +1,23 @@
...
-  const lonegerNumbersWithLimit = generateNumber(10, 45);
-
-  t.is(isSortedList(defaultLength), true);
-  t.is(isSortedList(longerNumbers), true);
-  t.is(isSortedList(lonegerNumbersWithLimit), true);
+	const defaultLength = generateNumber();
+	const longerNumbers = generateNumber(10);
+	const lonegerNumbersWithLimit = generateNumber(10, 45);
 
+	t.true(defaultLength.every(isSorted));
```
Comment (ragingwind) [inline src/test.ts]: As I mentioned above, we need a new test for sorting only

File: src/test.ts [hunk 5]
```diff
@@ -1,39 +1,23 @@
...
-    if (head > rest[0]) return false;
-
-    const [nextHead, ...nextRest] = rest;
-    return recursiveSortCheck(nextHead, nextRest);
-  }
-
-  const [head, ...rest] = list;
-  return recursiveSortCheck(head, rest);
+function isSorted(current: number, index: number, list: number[]) {
+	const nextEl: number | undefined = list[index + 1];
```
Comment (ragingwind) [inline src/test.ts]: nextEl? next would be fine

File: src/test.ts (hunk 2 above)
Comment (adhrinae) [reply src/test.ts]: ahh, I misunderstood what you mentioned before. I'll try it again.

**Comment by ragingwind** [comment]
Great, ! Thank for your hard work :)

**Comment by ragingwind** [comment]
One thing I would like recommend, test title changing should be another PR detached from this RP.

**Pull request merged**